src/cmd/flux-jobs.py

```diff
@@ -0,0 +1,202 @@
+        )
+        print(str)
+
+
+@flux.util.CLIMain(logger)
+def main():
+    parser = argparse.ArgumentParser()
+    # -a equivalent to -s "pending,running,inactive" and -u set to userid
+    parser.add_argument(
+        "-a", action="store_true", help="List all jobs for current user"
+    )
+    # -A equivalent to -s "pending,running,inactive" and -u set to "all"
+    parser.add_argument("-A", action="store_true", help="List all jobs for all users")
+    parser.add_argument(
+        "-c",
+        "--count",
+        type=int,
+        metavar="N",
+        default=1000,
+        help="Limit output to N jobs(default 1000)",
+    )
+    parser.add_argument(
+        "-s",
+        "--states",
+        type=str,
+        metavar="STATES",
+        default="pending,running",
+        help="List jobs in specific states(pending,running,inactive)",
+    )
+    parser.add_argument(
+        "--suppress-header",
+        action="store_true",
+        help="Suppress printing of header line",
+    )
+    parser.add_argument(
+        "-u",
+        "--userid",
+        type=int,
+        metavar="UID",
+        default=os.geteuid(),
+        help='Limit output to specific userid (Specify "all" for all users)',
+    )
+    parser.add_argument(
+        "-o",
+        "--format",
+        type=str,
+        metavar="FORMAT",
+        help="Specify output format using Python f-string formation",
+    )
+
+    args = parser.parse_args()
+
+    h = flux.Flux()
+
+    # Future optimization, reduce attrs based on what is in output
+    # format, to reduce potentially large return RPC.
+    attrs = [
+        "userid",
+        "priority",
+        "state",
+        "job-name",
+        "task-count",
+        "t_submit",
+        "t_depend",
+        "t_sched",
+        "t_run",
+        "t_cleanup",
+        "t_inactive",
+    ]
+
+    if args.a:
+        args.userid = os.geteuid()
+    if args.A:
+        args.userid = flux.constants.FLUX_USERID_UNKNOWN
+    if args.a or args.A:
+        args.states = "pending,running,inactive"
+
+    flags = 0
+    for state in args.states.split(","):
+        if state.lower() == "pending":
+            flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        elif state.lower() == "running":
+            flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+        elif state.lower() == "inactive":
+            flags |= flux.constants.FLUX_JOB_LIST_INACTIVE
+        else:
+            print("Invalid state specified", file=sys.stderr)
+            sys.exit(1)
+
+    if flags == 0:
+        flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+
+    rpc_handle = flux.job.job_list(h, args.count, attrs, args.userid, flags)
+    try:
+        jobs = flux.job.job_list_get(rpc_handle)
+    except EnvironmentError as e:
+        print("{}: {}".format("rpc", e.strerror), file=sys.stderr)
+        sys.exit(1)
+
+    if args.format:
+        output_format(args.format, jobs)
```

Comment:
Yeah, b/c it's Python f-strings it's hard to do this.  Since the user can specify virtually any formatting they want.  I think with `squeue` there's a finite set of output options with their `"type[:[.]size]" `format?

Reply:
I can think of a couple ways we could make this happen, mainly either extracting the variables with a parse `{\w*` or similar, or setting up a mock format that we use to determine which fields were accessed by the formatting process.  Either way this is probably a down-the-road thing but I think it's doable.

Reply:
given stephen's technique below of just generating a header via:

```
fmt.format(
id="JOBID",
state="STATE",
task_count="TASKS",
job_name="NAME",
username="USER",
runtime_fsd="TIME"
)
```

It's perhaps easier than I thought.  Ultimately, if the header doesn't look pretty, we leave it to the user to figure out?